spec/lib/metasploit/framework/credential_collection_spec.rb

```diff
@@ -120,6 +121,37 @@
       end
     end
 
+    context "when given a pass_file and user_file and password spray" do
+      let(:password) { nil }
+      let(:username) { nil }
+      let(:password_spray) { true }
+      let(:pass_file) do
+        filename = "pass_file"
+        stub_file = StringIO.new("password1\npassword2\n")
+        allow(File).to receive(:open).with(filename,/^r/).and_return stub_file
+
+        filename
+      end
+      let(:user_file) do
+        filename = "user_file"
+        stub_file = StringIO.new("user1\nuser2\nuser3\n")
+        allow(File).to receive(:open).with(filename,/^r/).and_yield stub_file
+
+        filename
+      end
+
+      specify  do
+        expect { |b| collection.each(&b) }.to yield_successive_args(
+          Metasploit::Framework::Credential.new(public: "user1", private: "password1"),
+          Metasploit::Framework::Credential.new(public: "user2", private: "password1"),
+          Metasploit::Framework::Credential.new(public: "user3", private: "password1"),
+          Metasploit::Framework::Credential.new(public: "user1", private: "password2"),
+          Metasploit::Framework::Credential.new(public: "user2", private: "password2"),
+          Metasploit::Framework::Credential.new(public: "user3", private: "password2"),
+        )
+      end
+    end
```

Comment:
Morning 👋 
I may have confused you with my explanation yesterday. I have added a diff with the exact changes that are needed for that test to pass 🤞 
```suggestion
    context "when given a pass_file and user_file and password spray" do
      let(:password) { nil }
      let(:username) { nil }
      let(:password_spray) { true }
      let(:user_file) do
        filename = "user_file"
        stub_file = StringIO.new("user1\nuser2\nuser3\n")
        allow(File).to receive(:open).with(filename,/^r/).and_return stub_file

        filename
      end
      let(:pass_file) do
        filename = "pass_file"
        stub_file = StringIO.new("password1\npassword2\n")
        allow(File).to receive(:open).with(filename,/^r/).and_yield stub_file

        filename
      end

      specify  do
        expect { |b| collection.each(&b) }.to yield_successive_args(
          Metasploit::Framework::Credential.new(public: "user1", private: "password1"),
          Metasploit::Framework::Credential.new(public: "user2", private: "password1"),
          Metasploit::Framework::Credential.new(public: "user3", private: "password1"),
          Metasploit::Framework::Credential.new(public: "user1", private: "password2"),
          Metasploit::Framework::Credential.new(public: "user2", private: "password2"),
          Metasploit::Framework::Credential.new(public: "user3", private: "password2"),
        )
      end
    end
```